test/sequential/test-repl-persistent-history.js

```diff
@@ -166,6 +172,18 @@ const tests = [{
   env: {},
   test: [UP],
   expected: [prompt, homedirErr, prompt, replDisabled, prompt]
+},
+{
+  env: { NODE_REPL_HISTORY: emptyHistoryPath,
+         NODE_REPL_HISTORY_FILE: emptyHistoryPath },
+  test: [UP],
+  expected: [prompt, sameHistoryFilePaths, prompt]
+},
+{
+  env: { NODE_REPL_HISTORY: defaultHistoryPath,
+         NODE_REPL_HISTORY_FILE: emptyHistoryPath },
+  test: [UP],
+  expected: [prompt, convertMsg, prompt]
```

Comment:
I'd recommend putting the tests down here though https://github.com/nodejs/node/pull/4108/files#diff-8b14be731a57727f74971922e046aab6R176 -- just a little more logical order imo, nothing big.